### ISSUE TITLE
ExternalInterfaceAttribute 

### ISSUE THREAD

**Issue by vladsch** [bridgedotnet/Bridge#1844]
In Bridge 15, interfaces implementation is chnaged to support all possible .NET cases
In 1.14, some cases are not supported, for example
```cs
public interface I {
    void Foo ();
}

public class A : I {
    public void Foo () {
        Console.WriteLine("A");
    }
}

public class B : A, I {
    void I.Foo () {
        Console.WriteLine("B");
    }
}
```

If create instance of `B` class and assign to `I` type variable then during compilation time we don't know which `Foo` method invocation should be generated (should be `B.Foo` which has `foo$1` name in javascript because `foo` name is already reserved by `A.Foo` method).

```cs
I i = new B();
i.Foo(); // should be translated to i.foo$1() but Bridge doesn't what exactly instance is assigned to `i` variable
```

Bridge 15 introduces interface member alias to avoid such collisions. For each interface member implementation will be generated alias like `I$foo` which will be referenced to implementation

```cs
I i = new B();
i.Foo(); // translated to `i.I$foo()`
```

If interface is external (created without Bridge type system) and its implementation is also external (no generated aliases) then such interface must marked by `[ExternalInterface]` to instruct Bridge that aliases are absent. In this case, Bridge will generate special code to get member in javascript (Bridge will check alias and direct name)
```js
i[Bridge.geti(i, "I$foo", "foo")]()
```

**Comment by vladsch** [comment]
http://forums.bridge.net/forum/bridge-net-pro/bugs/2719-members-on-external-interfaces-are-not-correctly-specified-in-the-javascript

**Comment by vladsch** [comment]
Implemented with #1025 issue
